### PR TITLE
Initial ARM Template

### PR DESCRIPTION
**Pull request by mwevill**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            "defaultValue": "[subscription().subscriptionId]"
+        },
+        "logAnalyticsResourceGroup": {
+            "type": "string"
+        },
+        "logAnalyticsWorkspaceName": {
+            "type": "string"
+        },
+        "configurationStorageConnectionString": {
+            "type": "string"
```
Comment (EwanNoble) [inline azure/template.json]: Connection strings should be of type 'securestring'

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        },
+        "loggingRedisConnectionString": {
+            "type": "string"
```
Comment (EwanNoble) [inline azure/template.json]: As above

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    "variables": {
+        "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+        "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName')))]",
+        "internalApiAppServiceName": "[concat(variables('resourceNamePrefix'), '-tprintapi-as')]",
+        "databaseName": "[concat(variables('resourceNamePrefix'), '-tpr-db')]"
+    },
+    "resources": [
+        {
+            "apiVersion": "2017-08-01",
+            "name": "InternalApiAppInsights",
```
Comment (EwanNoble) [inline azure/template.json]: Refer to here for deployment name formatting https://github.com/SkillsFundingAgency/das-platform-building-blocks/#property-name-formatting

Comment (mwevill) [reply azure/template.json]: Resolved

File: azure/template.json (hunk 2 above)
Comment (mwevill) [reply azure/template.json]: Resolved

File: azure/template.json [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                    "customHostName": {
+                        "value": "[parameters('internalApiCustomHostname')]"
+                    },
+                    "certificateThumbprint": {
+                        "value": "[reference(resourceId(parameters('sharedAppServicePlanResourceGroup'), 'Microsoft.Web/certificates', parameters('internalApiCertificateName')), '2016-03-01').Thumbprint]"
+                    }
+                }
+            },
+            "dependsOn": [
+                "InternalApiAppInsights"
```
Comment (EwanNoble) [inline azure/template.json]: This needs to be updated as well with the new name

**Pull request merged**